util/progress/progressui/printer.go

```diff
@@ -28,13 +29,21 @@ type lastStatus struct {
 
 type textMux struct {
 	w         io.Writer
+	printer   vertexPrinter
 	current   digest.Digest
 	last      map[string]lastStatus
 	notFirst  bool
 	nextIndex int
 	desc      string
 }
 
+func newTextMux(w io.Writer, printer vertexPrinter) *textMux {
+	if printer == nil {
+		printer = (*textPrinter)(nil)
```

Comment:
I can imagine this construct being confusing for the readers. As this is all private, let's avoid it and ensure the caller always does the right thing.

Reply:
Yea sure. I'll change this.